### PR TITLE
[5.8] hyphen-case the Syslog program name

### PR DESCRIPTION
**Pull request by barryvdh**
Follow existing program name conventions to have more readable logs. When using spaces, it looks kinda weird in Papertrail. Not sure about other services.
Eg. 'My Cool App' would be: my-cool-app, similar to other services (systemd-logind etc)

Papertrail also uses some logic to filter or group by name, which fails when using spaces..

### CODE REVIEW

**Comment by mfn** [comment]
👍 but rather in the next major release, could be too surprising for users of the current release.

File: src/Illuminate/Log/LogManager.php [hunk 1]
```diff
@@ -285,7 +285,7 @@ protected function createSyslogDriver(array $config)
     {
         return new Monolog($this->parseChannel($config), [
             $this->prepareHandler(new SyslogHandler(
-                $this->app['config']['app.name'], $config['facility'] ?? LOG_USER, $this->level($config))
+                snake_case($this->app['config']['app.name'], '-'), $config['facility'] ?? LOG_USER, $this->level($config))
```
Comment (lucasmichot) [inline src/Illuminate/Log/LogManager.php]: You can maybe directly use `Str::snake` ? No need to use a helper for that

**Comment by barryvdh** [comment]
Could do that, if Taylor let's me know I'll change it (can't properly change the base withou adding more commits for some reason)

**Comment by barryvdh** [comment]
Thanks!

**Pull request merged**